src/charm.py

```diff
@@ -73,19 +73,27 @@ def __init__(self, *args):
             self._name,
             resource_reqs_func=self._resource_reqs_from_config,
         )
+        self.framework.observe(self.resources_patch.on.patch_failed, self._on_k8s_patch_failed)
+
+        self.ingress = IngressPerUnitRequirer(self, relation_name="ingress", port=self._port)
+        self.framework.observe(self.ingress.on.ready_for_unit, self._on_ingress_ready)
+        self.framework.observe(self.ingress.on.revoked_for_unit, self._on_ingress_revoked)
```

Comment:
Just to make the code more legible... what about placing all the `self.framework.observe` all together?

Reply:
There was some back and forth about this.
Iirc we concluded to split them and have the `observe` right next to the lib that emits those events.